app/models/item_status.rb

```diff
@@ -52,15 +52,17 @@ def approval_time
 
   def approve!(user)
     @request.send_to_symphony_now!(barcodes: [@id])
-    reload_request # reloading to get any attributes saved to the database above
-    return unless symphony_item_successful?
-    self.status_object = {
-      approved: true,
-      approval_time: Time.zone.now.to_s,
-      approver: user
-    }.with_indifferent_access
-    @request.approval_status = :approved if @request.all_approved?
-    @request.save!
+    @request.with_lock do
+      @request.request_status_data ||= {}
+      next unless symphony_item_successful?
```

Comment:
Yes, it's to get out of the lock (it used to `return`, but I don't believe that works the same in this case).